src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nodeForIndex = findNode(index);
+            Node<T> pred = nodeForIndex.prev;
+            Node<T> newNode = new Node<>(pred, value, nodeForIndex);
+            nodeForIndex.prev = newNode;
+            if (pred == null) {
+                head = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> indexNode = head;
+        for (int i = 0; i < index; i++) {
+            indexNode = indexNode.next;
+        }
+        T result = indexNode.item;
+        indexNode.item = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> headNode = head;
+        while (headNode != null) {
+            if ((headNode.item == object)
+                    || (headNode.item != null && headNode.item.equals(object))) {
+                unlink(headNode);
+                return true;
+            }
+            headNode = headNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addFirst(Node<T> node) {
+        head = node;
+        tail = node;
+        size++;
+    }
+
+    private Node<T> findNode(int index) {
+        Node<T> indexNode;
+        if (index < size / 2) {
+            indexNode = head;
+            for (int i = 0; i < index; i++) {
+                indexNode = indexNode.next;
+            }
+        } else {
+            indexNode = tail;
+            for (int i = size - 1; i > index; i--) {
+                indexNode = indexNode.prev;
+            }
+        }
+        return indexNode;
+    }
+
+    private T unlink(Node<T> node) {
+        final T element = node.item;
+        if (node.prev == null && node.next == null) {
+            node.item = null;
+        }
+        if (node.prev != null && node.next == null) {
+            tail = node.prev;
+            node.prev.next = null;
+        }
+        if (node.prev == null && node.next != null) {
+            head = node.next;
+            node.next.prev = null;
+        }
+        if (node.prev != null && node.next != null) {
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+        }
```

Comment:
let's make it through operator case of for one of conditions: node.prev == null or node.next == null.